pkg/controller/workload/workload_processor.go

```diff
@@ -401,6 +425,11 @@ func (p *Processor) handleWorkload(workload *workloadapi.Workload) error {
 	p.WorkloadCache.AddOrUpdateWorkload(workload)
 	p.storeWorkloadPolicies(workload.GetUid(), workload.GetAuthorizationPolicies())
 
+	// update kmesh localityCache
+	if p.nodeName == workload.GetNode() {
+		p.locality.SetLocality(p.nodeName, workload.GetClusterId(), workload.GetNetwork(), workload.GetLocality())
```

Comment:
not sure i understand why only setLocality when p.nodeName == workload.GetNode()

Reply:
Because the kmesh process retrieves information about all workloads from xDS, and the locality associated with the kmesh process should be carried by the workloads on the same node, we only update locality information from workloads that have the same nodeName as the current one.